bleak/backends/corebluetooth/CentralManagerDelegate.py

```diff
@@ -146,11 +147,11 @@ async def connect_(self, peripheral: CBPeripheral, timeout=10.0) -> bool:
         self._connection_state = CMDConnectionState.PENDING
         self.central_manager.connectPeripheral_options_(peripheral, None)
 
-        start = time.time()
-        while self._connection_state == CMDConnectionState.PENDING:
-            await asyncio.sleep(0)
-            if time.time() - start > timeout:
-                raise TimeoutError("Failed to connect in %d seconds" % timeout)
+        try:
+            await asyncio.wait_for(self._connection_state_changed.wait(), timeout=timeout)
+        except TimeoutError:
```

Comment:
This needs to be `asyncio.TimeoutError` rather than the builtin `TimeoutError`.

Reply:
Great catch!